repository/src/main/java/org/apache/atlas/repository/store/graph/AtlasTypeDefGraphStore.java

```diff
@@ -1200,4 +1200,84 @@ private void tryTypeCreation(AtlasTypesDef typesDef, AtlasTransientTypeRegistry
             }
         }
     }
+    public void isIndexCreationFailed(AtlasTypesDef typesDef) throws AtlasBaseException{
```

Comment:
@hitarthtalavia  Please read [DRY](https://wiki.c2.com/?DontRepeatYourself).
Also I would suggest to rewrite this code block with below suggestion ( feel free to write your own code though )
```
    private boolean isIndexCreatedSuccessfully(List<? extends AtlasBaseTypeDef> typedefs) {
        return typedefs.stream().allMatch(typedef -> typedef.isIndexCreated());
    }

    public boolean isIndexCreated(AtlasTypesDef typesDef) {
        return isIndexCreatedSuccessfully(typesDef.getEntityDefs())
                && isIndexCreatedSuccessfully(typesDef.getEnumDefs())
                && isIndexCreatedSuccessfully(typesDef.getStructDefs())
                && isIndexCreatedSuccessfully(typesDef.getClassificationDefs())
                && isIndexCreatedSuccessfully(typesDef.getRelationshipDefs())
                && isIndexCreatedSuccessfully(typesDef.getBusinessMetadataDefs());

    }
```